sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -920,7 +931,7 @@ private[execution] object AggregateProcessor {
     // Create the projections.
     val initialProjection = newMutableProjection(
       initialValues,
-      Seq(SizeBasedWindowFunction.n))()
+      Seq(partitionSize))()
```

Comment:
@yhuai One of the `SizeBasedWindowFunction`s (NTILE) uses the `partitionSize` during initialization. So we need have it available as an input attribute during initialization.